batavia/modules/_compile/tokenizer.js

```diff
@@ -0,0 +1,1095 @@
+               } else if (c == 'e' || c == 'E') {
+                   return tok.exponent(c);
+               } else if (c == 'j' || c == 'J') {
+                   return tok.imaginary();
+               } else if (nonzero) {
+                   tok.done = E_TOKEN;
+                   tok.tok_backup(c);
+                   return [ERRORTOKEN, tok.start, tok.cur, 8];
+               }
+           }
+       } else {
+           // Decimal
+           do {
+               c = tok.tok_nextc();
+           } while (isdigit(c));
+           return tok.fraction(c);
+       }
+       tok.tok_backup(c);
+       p_start = tok.start;
+       p_end = tok.cur;
+       return [NUMBER, p_start, p_end];
+   }
+
+ var result = tok.letter_quote(c);
+ if (result !== batavia.builtins.None) {
+     return result;
+ }
+
+ // Line continuation
+ if (c == '\\') {
+     c = tok.tok_nextc();
+     if (c != '\n') {
+         tok.done = E_LINECONT;
+         tok.cur = tok.inp;
+         return [ERRORTOKEN, p_start, p_end, 9];
+     }
+     tok.cont_line = 1;
+     return tok.again(); // Read next line
+ }
+
+ // Check for two-character token
+ var c2 = tok.tok_nextc();
+ var token = PyToken_TwoChars(c, c2);
+ if (token != OP) {
+     var c3 = tok.tok_nextc();
+     var token3 = PyToken_ThreeChars(c, c2, c3);
+     if (token3 != OP) {
+         token = token3;
+     } else {
+         tok.tok_backup(c3);
+     }
+     p_start = tok.start;
+     p_end = tok.cur;
+     return [token, p_start, p_end];
+ }
+ tok.tok_backup(c2);
+
+ // Keep track of parentheses nesting level
+ switch (c) {
+ case '(':
+ case '[':
+ case '{':
+     tok.level++;
+     break;
+ case ')':
+ case ']':
+ case '}':
+     tok.level--;
+     break;
+ }
+
+ // Punctuation character
+ p_start = tok.start;
+ p_end = tok.cur;
+ return [PyToken_OneChar(c), p_start, p_end];
+};
+
+Tokenizer.prototype.parse_identifier = function(c) {
+var tok = this;
+var nonascii = 0;
+// Process b"", r"", u"", br"" and rb""
+var saw_b = 0;
+var saw_r = 0;
+var saw_u = 0;
+var saw_f = 0;
+while (1) {
+    if (!(saw_b || saw_u || saw_f) && (c == 'b' || c == 'B'))
+        saw_b = 1;
+    // Since this is a backwards compatibility support literal we don't
+    //   want to support it in arbitrary order like byte literals.
+    else if (!(saw_b || saw_u || saw_r || saw_f) && (c == 'u' || c == 'U'))
+        saw_u = 1;
+    // ur"" and ru"" are not supported
+    else if (!(saw_r || saw_u) && (c == 'r' || c == 'R'))
+        saw_r = 1;
+    else if (!(saw_f || saw_b || saw_u) && (c == 'f' || c == 'F'))
+        saw_f = 1;
+    else
+        break;
+    c = tok.tok_nextc();
+    if (c == '"' || c == '\'') {
+        return tok.letter_quote(c);
+    }
+}
+while (is_potential_identifier_char(c)) {
+    if (c >= 128) {
+        nonascii = 1;
+    }
+    c = tok.tok_nextc();
+}
+tok.tok_backup(c);
+if (nonascii && !verify_identifier(tok)) {
+    return [ERRORTOKEN, p_start, p_end, 10];
+}
+
+  // async/await parsing block.
```

Comment:
It is. We don't support `async` yet, but we will, and then we'll need this block of code to handle it.